js/app.js

```diff
@@ -0,0 +1,29 @@
+"use strict";
+
+(function(){
+  angular
+  .module( "songs", [
+    "ui.router",
+    "ngResource"
+  ])
+  .config([
+    "$stateProvider",
+    RouterFunction
+  ]);
+
+  function RouterFunction($stateProvider){
+    $stateProvider
+    .state("index", {
+      url: "/",
+      templateUrl: "index.html",
+  
+    })
+    .state("songIndex", {
+      url: "/songs",
+      templateUrl: "js/songs/index.html",
+      controller: "SongIndexController",
+      controllerAs: "SongIndexViewModel"
+    })
+  }
+
+}());
```

Comment:
Before diving into anything like directives, I encourage you to flesh out your CRUD functionality, starting with show. I'm guessing that you're getting practice with that in your project though! Let me know if that's something you'd like to walk through during a 1:1 or office hours.